go-controller/pkg/util/node_annotations.go

```diff
@@ -464,6 +464,20 @@ func NodeTransitSwitchPortAddrAnnotationChanged(oldNode, newNode *corev1.Node) b
 	return oldNode.Annotations[ovnTransitSwitchPortAddr] != newNode.Annotations[ovnTransitSwitchPortAddr]
 }
 
+func NodeZoneClusterChanged(node1, node2 *corev1.Node) bool {
```

Comment:
@flavio-fernandes what if we pass in the topology and whether the node is local or not as method arguments; and then we can do all the appropriate checks in this method.

We could throw in the check for NodeNetworkIDAnnotationChanged so we vasically we do all the IC related checks here

Reply:
Perhaps the base network controller would be a better place for such a method though.

Reply:
ah, make sense.  There is one codepath that did not care whether node is localzone or not. I will check that.

Reply:
@jcaamano I made the changes as you suggested. PTAL and see if that is easier to follow now.

Reply:
> Perhaps the base network controller would be a better place for such a method though.

I put that in util because all it uses is `util` stuff. If you strongly disagree, we can do the move it

Reply:
> > Perhaps the base network controller would be a better place for such a method though.
> 
> I put that in util because all it uses is `util` stuff. If you strongly disagree, we can do the move it

So one of the reasons I suggested this to be done in the base network controller is because this now deals with more information (network name/topology...) than just node annotations and you would have more reasonable/easy access to that information there but not a strong opinion.